equinix/data_source_network_device.go

```diff
@@ -0,0 +1,759 @@
+	}
+	if err := d.Set(neDeviceSchemaNames["PurchaseOrderNumber"], primary.PurchaseOrderNumber); err != nil {
+		return fmt.Errorf("error reading PurchaseOrderNumber: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["RedundancyType"], primary.RedundancyType); err != nil {
+		return fmt.Errorf("error reading RedundancyType: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["RedundantUUID"], primary.RedundantUUID); err != nil {
+		return fmt.Errorf("error reading RedundantUUID: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["TermLength"], primary.TermLength); err != nil {
+		return fmt.Errorf("error reading TermLength: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["AdditionalBandwidth"], primary.AdditionalBandwidth); err != nil {
+		return fmt.Errorf("error reading AdditionalBandwidth: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["OrderReference"], primary.OrderReference); err != nil {
+		return fmt.Errorf("error reading OrderReference: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["InterfaceCount"], primary.InterfaceCount); err != nil {
+		return fmt.Errorf("error reading InterfaceCount: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["CoreCount"], primary.CoreCount); err != nil {
+		return fmt.Errorf("error reading CoreCount: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["IsSelfManaged"], primary.IsSelfManaged); err != nil {
+		return fmt.Errorf("error reading IsSelfManaged: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["Interfaces"], flattenNetworkDeviceInterfaces(primary.Interfaces)); err != nil {
+		return fmt.Errorf("error reading Interfaces: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["VendorConfiguration"], primary.VendorConfiguration); err != nil {
+		return fmt.Errorf("error reading VendorConfiguration: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["UserPublicKey"], flattenNetworkDeviceUserKeys([]*ne.DeviceUserPublicKey{primary.UserPublicKey})); err != nil {
+		return fmt.Errorf("error reading VendorConfiguration: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["ASN"], primary.ASN); err != nil {
+		return fmt.Errorf("error reading ASN: %s", err)
+	}
+	if err := d.Set(neDeviceSchemaNames["ZoneCode"], primary.ZoneCode); err != nil {
+		return fmt.Errorf("error reading ZoneCode: %s", err)
+	}
+	if secondary != nil {
+		if v, ok := d.GetOk(neDeviceSchemaNames["Secondary"]); ok {
+			secondaryFromSchema := expandNetworkDeviceSecondary(v.([]interface{}))
+			secondary.LicenseFile = secondaryFromSchema.LicenseFile
+		}
+		if err := d.Set(neDeviceSchemaNames["Secondary"], flattenNetworkDeviceSecondary(secondary)); err != nil {
+			return fmt.Errorf("error reading Secondary: %s", err)
+		}
+	}
+	if primary.ClusterDetails != nil {
+		if v, ok := d.GetOk(neDeviceSchemaNames["ClusterDetails"]); ok {
+			clusterDetailsFromSchema := expandNetworkDeviceClusterDetails(v.([]interface{}))
+			primary.ClusterDetails.Node0.LicenseFileId = clusterDetailsFromSchema.Node0.LicenseFileId
```

Comment:
This section might not be required in the data source because in the API response, we never provide license token and license file id back.